src/Network/Transport/TCP.hs

```diff
@@ -1770,9 +1761,9 @@ socketToEndPoint (EndPointAddress ourAddress) theirAddress reuseAddr noDelay kee
           N.connect sock (N.addrAddress addr)
         mapIOException failed $ do
           sendMany sock
-                   (encodeInt32 theirEndPointId : prependLength [ourAddress])
+                   (encodeWord32 theirEndPointId : prependLength [ourAddress])
           recvInt32 sock
-      case tryToEnum response of
+      case decodeConnectionRequestResponse response of
```

Comment:
The `recvInt32 sock` above appears to be the last use of `recvInt32` and it seems to be being used at the type `Word32`. So I could also be replaced by `recvWord32` and then we can stop exporting `recvInt32` from the `.Internal` module entirely.